Growatt: migrate actions to dedicated action pages (#46138)
home-assistant/home-assistant.github.io: source/_actions/growatt_server.update_time_segment.markdown

3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the search box, search for and select **Growatt: Update time segment**.
6. Choose the **Device**, then set the segment values you want.
7. Select **Save**.

This action does not support targets. In the UI, you are not prompted to choose an area, device, entity, or label.

### Options in the UI

{% options_ui %}
Device:
  description: The Growatt MIN inverter to update.
  required: true
Segment ID:
  description: The time segment to configure (1 to 9).
  required: true
Battery mode:
  description: "The energy priority mode for this segment. One of `load_first` (power your home from solar or battery first), `battery_first` (charge the battery from solar or grid first), or `grid_first` (export to the grid first, discharging the battery if needed)."
  required: true
Start time:
  description: The start time for the segment, in `HH:MM` format.
  required: true
End time:
  description: The end time for the segment, in `HH:MM` format.
  required: true
Enabled:

>   description: Whether this time segment is active.
>   required: true

Reviewer: The YAML options list a default value for this, so it should be marked as optional:

```suggestion
  description: Whether this time segment is active. Default: false.
  required: false
```

action: |
  action: growatt_server.update_time_segment
  data:
    device_id: a1b2c3d4e5f6
    segment_id: 1
    batt_mode: battery_first
    start_time: "00:00"
    end_time: "06:00"
    enabled: true
{% endexample %}

This charges the battery between midnight and 6 AM using the first time segment.

### Options in YAML

{% options_yaml %}
device_id:
  description: >
    The Growatt MIN inverter to update.
  required: true
  type: string
segment_id:
  description: >
    The time segment to configure (1 to 9).
  required: true
  type: integer
batt_mode:
  description: >
    The energy priority mode for this segment. One of `load_first`
    (power your home from solar or battery first), `battery_first`
    (charge the battery from solar or grid first), or `grid_first`
    (export to the grid first, discharging the battery if needed).
  required: true
  type: string
start_time:
  description: >
    The start time for the segment, in `HH:MM` format.
  required: true
  type: time
end_time:
  description: >
    The end time for the segment, in `HH:MM` format.
  required: true
  type: time
enabled:
  description: >
    Whether this time segment is active.
  required: true
  type: boolean

>   default: false

Reviewer: If this has a default value, list it as not required:

```suggestion
  required: false
  type: boolean
  default: false
```

{% endoptions_yaml %}

## Good to know

The battery mode controls when and why discharging happens. The actual discharge rate is controlled by the **Discharge power** number entity (0 to 100%).

{% include actions/try_it.md %}

> {% include actions/more_examples.md %}

Reviewer: Delete this when there's no examples:

```suggestion
```
